### ISSUE TITLE
Customization points, impl/examples for kokkos implementations

### ISSUE THREAD

**Issue by fnrizzi** [kokkos/stdBLAS#125]
Here is my plan, which I also shared in slack: 
(1) finish all customization points (I think I am pretty close to doing that) 
(2) focus on impl/examples for kokkos. Would it make sense to have all customizations and examples for kokkos moved to a separate repo that uses stdBLAS rather than leaving all code inside as it is now? This would allow to mimic the actual setup external libraries would need to do, and potentially can use this new thing as "template" to follow. 
I can then "convert" the example currently inside `examples/kokkos-based` to actual tests (this would address [comment](https://github.com/kokkos/stdBLAS/pull/123#discussion_r753745673)) and then proceed accordingly. So I would add implementations and tests inside this new repo. 

So basically, what I am proposing is: 
(1) move the kokkos impl to a separate repo (call it: `stdBLAS-kokkosimpl` or similar) that uses stdBLAS. 
(2) work on the actual kokkos-based implementations and add corresponding tests 

@mhoemmen @crtrott let me know what you think of this.

**Comment by mhoemmen** [comment]
@fnrizzi Thanks for working on this!  You don't have to do all the customization points in a single PR.  It's up to you, though : - )

**Comment by fnrizzi** [comment]
hi @mhoemmen !
- by customization points I meant adding to stdBLAS the code necessary to *allow* customization, since I have pretty much done BLAS2, and part of BLAS3 in the outstanding PRs, I think I only have a few left to finish BLAS3. 

- so just to clarify, you agree with my idea of moving all kokkos-related stuff to a separate repo?

**Comment by mhoemmen** [comment]
@fnrizzi wrote:
> ...you agree with my idea of moving all kokkos-related stuff to a separate repo?

I think that's a perfectly fine plan : - ) .  If that separate repo depends on both Kokkos and stdBLAS, then it should be possible to set up automated testing.

**Comment by fnrizzi** [comment]
thanks @mhoemmen for the feedback. 

@crtrott 
- do you agree with this too? 
- if so, would you mind helping me since I cannot make a repo under kokkos github organization ? 
I would vote to call it `stdBLAS-kokkosimpl` but am happy

**Comment by crtrott** [comment]
Not sure if we want to do this right now. If we move it out the tpl stuff wouldn't go into a new repo anyway but rather into kokkos-kernels itself. But for now I like the option to just bend around std::execution::par etc. to KokkosKernels which kinda requires it to be here? Hm maybe not. But lets keep this here until we have this actually all working, than we can talk with the kokkos-kernels folks about how to proceed.

**Comment by fnrizzi** [comment]
sounds good
